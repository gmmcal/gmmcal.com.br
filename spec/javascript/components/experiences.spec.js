import { mount, shallowMount } from '@vue/test-utils'
import Experiences from '@/components/experiences'
import Experience from '@/components/experience'
import create from '../settings/utils'

describe('Experiences', () => {
  let wrapper

  beforeEach(() => {
    window.gon = { experiences: create('experiences') }

    wrapper = mount(Experiences)
  })

  it('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  it('has data properties', () => {
    const expected = ['experiences', 'I18n']
    const received = Object.keys(wrapper.vm.$data)
    expect(received).toEqual(expected)
  })

  it('renders correctly', () => {
    expect(wrapper).toMatchSnapshot()
  })

  describe('Child components', () => {
    it('includes Experience', () => {
      const child_component = wrapper.find(Experience)
      expect(child_component.isVueInstance()).toBeTruthy()
    })

    it('renders 2 Experience objects', () => {
      wrapper = shallowMount(Experiences)
      const experiences = wrapper.findAll(Experience)
      expect(experiences.length).toBe(2)
    })
  })
})
