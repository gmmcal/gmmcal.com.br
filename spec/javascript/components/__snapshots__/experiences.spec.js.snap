// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Experiences renders correctly 1`] = `
<section id="experience" class="experience">
  <div class="container">
    <div class="heading">
      <h2>Work Experience</h2>
      <p>Companies that I have worked for</p>
    </div>
    <div class="row work-details">
      <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
        <div class="year">Jan / 2015<br>Jan / 2016</div>
      </div>
      <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
        <div class="arrowpart"></div>
        <div class="ex-con">
          <h4><span>Amazing company</span> - <span><span>Great city</span> - <span>Some country</span></span></h4>
          <h5>Ninja developer</h5>
          <div class="description">
            <p>Lorem ipsum nulus dolor</p>
          </div>
        </div>
      </div>
    </div>
    <div class="row work-details">
      <div class="col-xs-12 col-sm-12 col-md-3 col-lg-2">
        <div class="year">Jan / 2015<br>Jan / 2016</div>
      </div>
      <div class="col-xs-12 col-sm-12 col-md-9 col-lg-10 right-area">
        <div class="arrowpart"></div>
        <div class="ex-con">
          <h4><span>Amazing company</span> - <span><span>Great city</span> - <span>Some country</span></span></h4>
          <h5>Ninja developer</h5>
          <div class="description">
            <p>Lorem ipsum nulus dolor</p>
          </div>
        </div>
      </div>
    </div>
  </div>
</section>
`;
